### ISSUE TITLE
Error when configuring server for MQTT and reboot

### ISSUE THREAD

**Issue by bocklucas** [appdevelopmentandsuch/AnotherTempSensor#10]
**Summary**

The following error gets thrown some times when after an inital reboot after being configured to use the MQTT service.

```
--------------- CUT HERE FOR EXCEPTION DECODER ---------------

Soft WDT reset

>>>stack>>>

ctx: cont
sp: 3ffffd80 end: 3fffffc0 offset: 01a0
3fffff20:  3ffee900 3ffe8682 3fff0255 40204a48  
3fffff30:  3fff0594 3fff0224 3fff02ad 3fff0594  
3fffff40:  3fff0624 4f704800 3fff0614 3fff0594  
3fffff50:  3fffdaa0 00000400 3ffeeae4 3ffeec68  
3fffff60:  3fffdad0 3ffe862d 3ffee8fc 4020486c  
3fffff70:  feefeffe 3ffefe1c 3ffefea5 3fff018c  
3fffff80:  3fff021c feefef00 3fff020c 3fff018c  
3fffff90:  feefefa0 feefeffe feefeffe feefeffe  
3fffffa0:  feefeffe 00000000 3ffeec28 4020c9c0  
3fffffb0:  feefeffe feefeffe 3ffe84e4 40100f29  
<<<stack<<<

--------------- CUT HERE FOR EXCEPTION DECODER ---------------
```

While this isn't ideal at all, its not inhibiting the device thus far and seems to only happen every so often. When it does happen, the device simply resets and continues to function as expected.

**Comment by bocklucas** [comment]
Tentatively closing this, I believe the issue to be the software watch dog would restart the device due to deadlock on attempting to connect to the MQTT broker. 

I have put in a fix that makes the device attempt a certain number of times to connect before resetting.

**Comment by bocklucas** [comment]
Closed in #9
